OptimizelySDK/Odp/LruCache.cs

```diff
@@ -0,0 +1,128 @@
+﻿/* 
+ * Copyright 2022, Optimizely
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using OptimizelySDK.Logger;
+using System;
+using System.Collections.Specialized;
+
+namespace OptimizelySDK.Odp
+{
+    public class LruCache<T> : ILruCache<T>
+    {
+        public const int DEFAULT_MAX_SIZE = 10000;
+        public const int DEFAULT_TIMEOUT_SECONDS = 600;
+
+        private readonly ILogger _logger;
+        private readonly object _mutex = new object();
+        private readonly int _maxSize;
+        private readonly long _timeoutMilliseconds;
+        private readonly OrderedDictionary _orderedDictionary = new OrderedDictionary();
+
+        public LruCache() : this(DEFAULT_MAX_SIZE, DEFAULT_TIMEOUT_SECONDS, null) { }
+
+        public LruCache(int maxSize, int timeoutSeconds, ILogger logger = null)
+        {
+            _maxSize = maxSize < 0 ? default : maxSize;
```

Comment:
Let's set to zero if maxSize is negative. For consistency with other SDKs.
```suggestion
            _maxSize = maxSize < 0 ? 0 : maxSize;
```